core/src/test/scala/unit/kafka/log/LogManagerTest.scala

```diff
@@ -1303,6 +1303,27 @@ class LogManagerTest {
         createLeaderAndIsrRequestForStrayDetection(present),
         onDisk.map(mockLog(_))).toSet)
   }
+
+  @Test
+  def testLock(): Unit = {
+    val tmpLogDir = TestUtils.tempDir()
+    val tmpLogManager = createLogManager(Seq(tmpLogDir))
+    tmpLogManager.startup(Set.empty)
+
+    // ${tmpLogDir}.lock is acquired by tmpLogManager
+    val fileLock = new FileLock(new File(tmpLogDir, ".lock"))
+    try {
+      assertFalse(fileLock.tryLock())
+    } finally {
+      fileLock.destroy()
+    }
+
+    // ${tmpLogDir}.lock is removed after shutdown
+    tmpLogManager.shutdown()
+    val f = new File(tmpLogDir, ".lock")
+    assertFalse(f.exists())
+    Utils.delete(tmpLogDir)
```

Comment:
We should also wrap the logmanager.shutdown and tmpLogDir deletion in the `finally` block to make sure resources are released.